src/Tests/ImageBundleTest.php

```diff
@@ -0,0 +1,136 @@
+   */
+  protected $bundleStorage;
+
+  /**
+   * The test user.
+   *
+   * @var \Drupal\User\UserInterface
+   */
+  protected $adminUser;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    $entityManager = $this->container->get('entity.manager');
+    $this->bundleStorage = $entityManager->getStorage('media_bundle');
+    $this->testBundle = $this->bundleStorage->load('image');
+
+    $this->adminUser = $this->drupalCreateUser([
+        'view media',
+        'create media',
+        'update media',
+        'update any media',
+        'delete media',
+        'delete any media',
+        'access media overview',
+    ]);
+    $this->drupalLogin($this->adminUser);
+  }
+
+  /**
+   * Tests image media bundle creation from config files.
+   */
+  public function testMediaBundleCreationFromModule() {
+    $type_configuration = array(
+      'source_field' => 'field_image',
+      'gather_exif' => false
+    );
+
+    $this->assertTrue((bool) $this->testBundle, 'The media bundle from default configuration has been created in the database.');
+    $this->assertEqual($this->testBundle->get('label'), 'Image', 'Correct label detected.');
+    $this->assertEqual($this->testBundle->get('description'), 'Use Image for uploading locally hosted images.', 'Correct description detected.');
+    $this->assertEqual($this->testBundle->get('type'), 'image', 'Correct plugin ID detected.');
+    $this->assertEqual($this->testBundle->get('type_configuration'), $type_configuration, 'Type configuration correct.');
+    $this->assertEqual($this->testBundle->get('field_map'), [], 'Correct field map detected.');
+  }
+
+  /**
+   * Tests item creation and thumbnail
+   */
+  public function testMediaBundleItemCreation() {
+    // Define the media item name.
+    $name = $this->randomMachineName();
+    $image_files = $this->drupalGetTestFiles('image');
+    $testImage = File::create((array) current($image_files));
+    $file_path = $this->container->get('file_system')->realpath($testImage->getFileUri());
+    $edit = [
+      'name[0][value]' => $name,
+      'files[field_image_0]' => $file_path,
+    ];
+
+    // Save the image
+    $this->drupalPostForm('media/add/' . $this->testBundle->id(), $edit, t('Save and publish'));
+    $this->drupalPostForm(NULL, ['field_image[0][alt]' => $name], t('Save and publish'));
+    // Verify the title
+    $this->assertTitle($edit['name[0][value]'] . ' | Drupal');
+
+    // Let's retrieve the media id and corresponding media entity object
+    $media_id = $this->container->get('entity.query')->get('media')->execute();
+    $media_id = reset($media_id);
+    /** @var \Drupal\media_entity\MediaInterface $media */
+    $media = $this->container->get('entity_type.manager')
+        ->getStorage('media')
+        ->loadUnchanged($media_id);
+
+    // Now let's grab the image, thumbnail and the thumbnail URL
+    $field_name = 'field_image';
+    $file = $media->{$field_name}->entity;
```

Comment:
No need to create another var if you use it just once. `$media->field_image->entity` will do just fine and will be less obfuscated.